### PR TITLE
Move compile_sass tool to a w_common_tools package

### PR DESCRIPTION
**Pull request by danaoredson-wf**
## Motivation
w_common has some dependencies that are not desirable to expose to consumers. These appear to only be needed because of the compile_sass target. So separate out the compile_sass target into a separate project, so consumers can reduce their transient dependencies.
## Changes
* Adds a w_common_tools package
* Moves compile_sass to w_common_tools

NOTE: This is a major change.
#### Release Notes
Separate compile_sass into a new package to reduce transient dependencies for consumers.
### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: README.md [hunk 1]
```diff
@@ -11,20 +11,23 @@ includes the following:
   * `InvalidationMixin` mixin used to mark a class as requiring validation.
   * `JsonSerializable` interface to indicate that something can be serialized
   to JSON.
+
+# w_common_tools
+A collection of dev utilities that are not intended to be used in a deployable artifacts.
```
Comment (danaoredson-wf) [inline README.md:16]:
```suggestion
A collection of dev utilities that are not intended to be used in deployable artifacts.
```

File: .github/workflows/dart_ci.yml [hunk 2]
```diff
@@ -38,6 +38,26 @@ jobs:
         run: dart test
         if: always() && steps.install.outcome == 'success'
 
+      - id: install-tools
```
Comment (corwinsheahan-wf) [inline .github/workflows/dart_ci.yml:41]: Looks like the rest of these are not special cased
```suggestion
      - id: Install tools
```

File: .github/workflows/dart_ci.yml [hunk 3]
```diff
@@ -53,3 +73,13 @@ jobs:
       - name: Verify formatting
         run: dart format --output=none --set-exit-if-changed .
         if: always() && steps.install.outcome == 'success'
+
+      - id: install-tools
```
Comment (corwinsheahan-wf) [inline .github/workflows/dart_ci.yml:77]:
```suggestion
      - id: Install tools
```

File: .github/workflows/dart_ci.yml [hunk 4]
```diff
@@ -38,6 +38,26 @@ jobs:
...
 
+      - id: install-tools
+        name: Install w_common_tools dependencies
+        working-directory: ./w_common_tools
+        run: dart pub get
+
+      - name: Validate w_common_tools  dependencies
+        working-directory: ./w_common_tools
+        run: dart run dependency_validator
+        if: always() && steps.install-tools.outcome == 'success'
```
[the remark this answers is not in the document]
Comment (danaoredson-wf) [reply .github/workflows/dart_ci.yml:49]: Not sure. This was a copy-paste of the other targets.  I was wondering why the existing ones didn't have a `needs` clause.

File: w_common_tools/LICENSE [hunk 5]
```diff
@@ -0,0 +1,13 @@
+Copyright 2016 Workiva Inc.
```
[the remark this answers is not in the document]
Comment (danaoredson-wf) [reply w_common_tools/LICENSE:1]: I was thinking the same thing. If you also do, then I'll change it.

File: .github/workflows/dart_ci.yml (hunk 2 above)
Comment (danaoredson-wf) [reply .github/workflows/dart_ci.yml:41]: I believe this id is used in the steps.install-tools.outcome evaluations, so this wouldn't work.

However, it seems like these should all be using a `needs` clause instead. But I didn't want to make that change and be inconsistent.

File: .github/workflows/dart_ci.yml (hunk 3 above)
Comment (danaoredson-wf) [reply .github/workflows/dart_ci.yml:77]: Same response as above.

File: w_common_tools/LICENSE (hunk 5 above)
Comment (danaoredson-wf) [inline w_common_tools/LICENSE:1]:
```suggestion
Copyright 2022 Workiva Inc.
```

**Pull request merged**